packages/flutter_test/lib/src/binding.dart

```diff
@@ -373,6 +373,18 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
   /// [AutomatedTestWidgetsFlutterBinding] implementation to increase the
   /// current timeout, if any. See [AutomatedTestWidgetsFlutterBinding.addTime]
   /// for details.
+  ///
+  /// If your test code depends on the result of a task that did not complete,
```

Comment:
nit: maybe we should restructure this block a little bit for clarity and lead with "if you're experiencing a hanging test, <reasons why this may be>". I imagine, if you scan the docs for a solution, it is because of a hanging test and you don't know yet that it is because of an un-completed task.